jena-geosparql/src/main/java/org/apache/jena/geosparql/implementation/GeometryWrapper.java

```diff
@@ -1,1248 +1,1240 @@
+        }
+        return geometryDatatype;
+    }
+
+    /**
+     * Transformations between SRS and Units will increase the inaccuracy of the
+     * results.
+     *
+     * @param distance
+     * @param targetDistanceUnitsURI
+     * @return Buffer around GeometryWrapper according the provided distance.
+     * @throws FactoryException
+     * @throws MismatchedDimensionException
+     * @throws TransformException
+     * @throws UnitsConversionException
+     */
+    public GeometryWrapper buffer(double distance, String targetDistanceUnitsURI) throws FactoryException, MismatchedDimensionException, TransformException, UnitsConversionException {
+
+        //Check whether the source geometry is linear units for cartesian calculation. If not then transform to relevant UTM SRS GeometryWrapper.
+        Boolean isTargetUnitsLinear = UnitsRegistry.isLinearUnits(targetDistanceUnitsURI);
+
+        if (srsInfo.getUnitsOfMeasure().isLinearUnits() != isTargetUnitsLinear) {
+            //Source geometry and target units are not the same.
+            throw new UnitsConversionException("Buffer does not support conversion between linear and non-linear units.");
+        }
+
+        //Transform distance into current units.
+        double transformedDistance = UnitsOfMeasure.conversion(distance, targetDistanceUnitsURI, this.srsInfo.getUnitsOfMeasure().getUnitURI());
+
+        //Buffer the geometry by the transformed distance.
+        Geometry xyGeo = this.xyGeometry.buffer(transformedDistance);
+        DimensionInfo bufferedDimensionInfo = new DimensionInfo(dimensionInfo.getCoordinate(), dimensionInfo.getSpatial(), xyGeo.getDimension());
+        Geometry parsingGeo = GeometryReverse.check(xyGeo, this.srsInfo);
+        return new GeometryWrapper(parsingGeo, xyGeo, this.srsInfo.getSrsURI(), this.geometryDatatypeURI, bufferedDimensionInfo);
+    }
+
+    /**
+     *
+     * @return URI of the GeometryWrapper's UTM zone
+     * @throws FactoryException
+     * @throws MismatchedDimensionException
+     * @throws TransformException
+     */
+    public String getUTMZoneURI() throws FactoryException, MismatchedDimensionException, TransformException {
+
+        if (utmURI == null) {
+
+            //Find a point in the parsing geometry so can directly apply the SRS.
+            Point coord = parsingGeometry.getCentroid();
+            DirectPosition2D point = new DirectPosition2D(coord.getX(), coord.getY());
+
+            //Convert to WGS84. Use WGS84 and not CRS84 as assuming WGS8 is more prevalent.
+            CoordinateReferenceSystem wgs84CRS = SRSRegistry.getCRS(SRS_URI.WGS84_CRS);
+            MathTransform transform = MathTransformRegistry.getMathTransform(srsInfo.getCrs(), wgs84CRS);
+
+            DirectPosition wgs84Point = transform.transform(point, null);
+
+            //Find the UTM zone.
+            utmURI = SRSRegistry.findUTMZoneURIFromWGS84(wgs84Point.getOrdinate(0), wgs84Point.getOrdinate(1));
```

Comment:
[`SRSRegistry.findUTMZoneURIFromWGS84`](https://github.com/apache/jena/blob/6e4837ee51f7c1a7bc3dfa18a21abd4c3f94d1bb/jena-geosparql/src/main/java/org/apache/jena/geosparql/implementation/registry/SRSRegistry.java#L124) is for `latitude` and `longitude`. But in the opengis [docs](https://docs.geotools.org/latest/javadocs/org/opengis/geometry/DirectPosition.html#getOrdinate-int-) all that I could find is that `getOrdinate` returns the ordinate at the dimension N :disappointed_relieved: 

Just checking if we are sure that `0` is latitude, and `1` is longitude?

Reply:
In `WGS84` the first coordinate is the latitude and the second is the longitude. I've added a couple of tests using `WGS84` and `CRS84` (which reverses the coordinate order). The coordinates are both for the UK and fall within UTM zone 30N and the tests return this result.